app/controllers/SectionCompletedStateController.scala

```diff
@@ -0,0 +1,88 @@
+import models.mongo.{JourneyAnswers, JourneyStatus}
+import models.mongo.JourneyStatus.{Completed, InProgress}
+import models.{BenefitType}
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.libs.json.Json
+import play.api.mvc._
+import services.SectionCompletedService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.pages.SectionCompletedStateView
+import actions.TaxYearAction
+
+import java.time.Instant
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SectionCompletedStateController @Inject()(implicit val cc: MessagesControllerComponents,
+                                                authAction: AuthorisedAction,
+                                                view: SectionCompletedStateView,
+                                                errorHandler: ErrorHandler,
+                                                implicit val appConfig: AppConfig,
+                                                sectionCompletedService: SectionCompletedService,
+                                                ec: ExecutionContext
+                                               ) extends FrontendController(cc) with I18nSupport {
+
+  def form(): Form[Boolean] = YesNoForm.yesNoForm("sectionCompletedState.error.required")
+
+  def show(taxYear: Int, benefitType: BenefitType): Action[AnyContent] =
+    (authAction andThen TaxYearAction(taxYear, appConfig, ec)).async { implicit user =>
+          sectionCompletedService.get(user.user.mtditid, taxYear, benefitType.typeName).flatMap {
+            case Some(value) =>
+              value.data("status").validate[JourneyStatus].asOpt match {
+                case Some(JourneyStatus.Completed) =>
+                  Future.successful(Ok(view(form().fill(true), taxYear, benefitType)))
+
+                case Some(JourneyStatus.InProgress) =>
+                  Future.successful(Ok(view(form().fill(false), taxYear, benefitType)))
+
+                case _ => Future.successful(Ok(view(form(), taxYear, benefitType)))
+              }
+            case None => Future.successful(Ok(view(form(), taxYear, benefitType)))
+          }
+      }
+
+
+  def submit(taxYear: Int, benefitType: BenefitType): Action[AnyContent] = (authAction andThen TaxYearAction(taxYear, appConfig, ec)).async { implicit user =>
+    form()
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, taxYear, benefitType))),
+        answer => {
+         saveAndRedirect(answer, taxYear, benefitType, user.user.mtditid)
+        }
+      )
+  }
+
+  private def saveAndRedirect(answer: Boolean, taxYear: Int, benefitType: BenefitType, mtditid: String)(implicit hc: HeaderCarrier): Future[Result] = {
+    val status: JourneyStatus = if (answer) Completed else InProgress
+    val model = JourneyAnswers(mtditid, taxYear, benefitType.toString, Json.obj({
```

Comment:
Should be benefitType.typeName currently

However we might add new attributes to the BenefitType model so just hold off at the moment

Reply:
We might need to hear back off team-b first but will discuss monday